tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/ContextPropagation.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright file="ContextPropagation.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Kinesis
+{
+    internal static class ContextPropagation
+    {
+        private const string KinesisKey = "_datadog";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ContextPropagation));
+
+        public static void InjectTraceIntoRecords<TRecordsRequest>(IPutRecordsRequest request, SpanContext context)
```

Comment:
.NET performance optimization: Since we are already creating a duck type for the request object, we can get better performance by continuing to use a generic constraint each time we pass the object into another method

```suggestion
        public static void InjectTraceIntoRecords<TRecordsRequest>(TPutRecordsRequest request, SpanContext context)
            where TPutRecordsRequest : IPutRecordsRequest, IDuckType
```